### PR TITLE
V9.9.6 Stable merge 

### PR DESCRIPTION
**Pull request by glynhudson**
Thankyou to everyone who contributed, including:

- Input and feed list interface development by @adminde 
- Support multiple redis databases @karlp https://github.com/emoncms/emoncms/pull/1159/commits/67a36a28891adcb8c4784512d03fc9c27fb8dc38
- Fix clear feed data https://github.com/emoncms/emoncms/commit/2055eb3d807ac442848cf47679e0b77d57ac7e95
- add autocomplete to node name field @emrysr https://github.com/emoncms/emoncms/pull/1159/commits/e5c76a7ec266042c8ba839fa3f14200cfd3d6b1c
- fix phpfina caching resulting in clear data issue: https://github.com/emoncms/emoncms/pull/1159/commits/d977f456b40b6efbe41de45c8b57fe735e1eddf6
- Fix smothie graph https://github.com/emoncms/emoncms/pull/1159/commits/d471c07deff8c2a8472f34d0f80001ae910d99ab
- Feedwiter systemd support: https://github.com/emoncms/emoncms/pull/1159/commits/6f0ce0e81b33ce7d80818fa68a0b5570dacc4576
- JSON output fix @fcauwe  https://github.com/emoncms/emoncms/pull/1159/commits/17a4c78a6019b39c4984f9ac04d56cb7cdc78890
- Show status of services on admin page p1: https://github.com/emoncms/emoncms/pull/1159/commits/83314de34b524079da0ee98471987477f51b9732
- multi feed data and average end points @emrysr https://github.com/emoncms/emoncms/pull/1159/commits/a525bb4a9458076e233b1b62ee39c62f67edda8f & https://github.com/emoncms/emoncms/pull/1159/commits/5f995ab667690123e3de3c2ef80b8072048fa19c
 - theme menu improvements @adminde https://github.com/emoncms/emoncms/pull/1159/commits/ee0bcc51e179c32ff197410e3031851bf48d5d1e
- fix padding spikes: https://github.com/emoncms/emoncms/pull/1159/commits/fc2fa84fcc615642e9d9fc9f05c8198c6ee0a4fe
- Support gzip decompression of bulk input data @SeanDS  https://github.com/emoncms/emoncms/pull/1159/commits/f8e05da06359d4d02e54343c81ff5a9a175cf53c
- edit phpfina data feeds @alexandrecuer  https://github.com/emoncms/emoncms/pull/1159/commits/92d94660ed1aa1d739c2f8a8cf9096d07ea0e909
- remove colon in phpmqtt_input @borpin https://github.com/emoncms/emoncms/pull/1159/commits
- feed engine method documentation @alexandrecuer 
- virtual feeds not included in feed last updated group heading @emrysr  https://github.com/emoncms/emoncms/pull/1159/commits
- node name autocomplete modularisation @emrysr  https://github.com/emoncms/emoncms/pull/1159/commits/9ee7566544f76d1e5f312902585fa5bb14e985d7
- fix bargraph height @emrysr https://github.com/emoncms/emoncms/pull/1159/commits/656117fc2c89187e620bdca2b10ddd84f9de11e8
- phpmqtt_input connection error @emrysr https://github.com/emoncms/emoncms/pull/1159/commits/2de73b56ef3d53aa9073dc1aa1dd6acaad3499dd
- allow read access to read api end points in input list: https://github.com/emoncms/emoncms/pull/1159/commits/fc4a4d11ca77e3efc7f7df9ee857cc01f07ebb2a
- disable use of virtual feed as source for itself: https://github.com/emoncms/emoncms/pull/1159/commits/c206e219be7b4e57d8210dcc1e6bf7a0e89689fb
- show feed interval status with colour bars @emrysr https://github.com/emoncms/emoncms/pull/1159/commits/2365684d7cbbed0f10c158de862f92d358f7b961
- Add support for Find Local Device's module @emrysr  https://github.com/emoncms/find

### CODE REVIEW

**Comment by borpin** [comment]
Have you setup the git tags so it shows up as a version release 😄

**Comment by borpin** [comment]
Hey guys - no version tags.

**Comment by SeanDS** [comment]
Regarding "Support gzip decompression of bulk input data @SeanDS f8e05da" - there was a suggestion to also support the "compress" query argument in addition to "c", and make it case insensitive - I can add these to the merge request in the next day or two if you want to hold off on this release...

**Comment by borpin** [comment]
How on earth, did that last comment of mine end up here!

File: scripts/services/emoncms_mqtt/emoncms_mqtt.service [hunk 1]
```diff
@@ -33,15 +33,15 @@
 ###
 
 [Unit]
-Description=Emoncms mqtt_input script
+Description=Emoncms emoncms_mqtt script
 Wants=mosquitto.service mysql.service redis.service
```
Comment (borpin) [inline scripts/services/emoncms_mqtt/emoncms_mqtt.service]: This should be `redis-server.service`

**Comment by alexandrecuer** [comment]
Just trying to create virtual feeds with that PR 1159

It appears not to be possible....

just after adding the source feed, when i want to save, I got the following message box : "ERROR: Could not save processlist. Cannot use virtual feed as source" but I did not choose a virtual feed as a source

when I checkout to a older version, I have no problem

Did anybody  face a similar problem ?

Alex
virtual_feeds_save_error

**Pull request merged**